ImperatorToCK3/Source/CommonUtilities/History.h

```diff
@@ -0,0 +1,53 @@
+#ifndef HISTORY_H
+#define HISTORY_H
+
+
+#include "Parser.h"
+#include "Date.h"
+#include <map>
+
+
+
+struct SimpleFieldStruct {
+	std::string fieldName;
+	std::string setter;
+	std::optional<std::string> initialValue;
+};
+
+class SimpleField {
+public:
+	SimpleField() = default;
+	explicit SimpleField(const std::optional<std::string>& initialValue);
+	[[nodiscard]] std::optional<std::string> getValue(const date& date) const;
+	[[nodiscard]] const auto& getValueHistory() const { return valueHistory; }
+
+	void setInitialValue(const std::optional<std::string>& newValue) { initialValue = newValue; }
+	void addValueToHistory(const std::string& value, const date& date) { valueHistory[date] = value; }
+
+private:
+	std::map<date, std::string, std::greater<>> valueHistory;
+	std::optional<std::string> initialValue = std::nullopt;
+};
+
+
+
+class DatedHistoryEntry: public commonItems::parser {
```

Comment:
This isn't used anywhere. Is it a future item or something that's redundant?

Reply:
Or do you just need more tests?

Reply:
It's used inside the Factory for `<date> = { stuff }` blocks.